app/routes/datasets/versions.py

```diff
@@ -232,13 +231,18 @@ async def append_to_version(
     # For the background task, we only need the new source uri from the request
     input_data = {"creation_options": deepcopy(default_asset.creation_options)}
     input_data["creation_options"]["source_uri"] = request.source_uri
+    # check if layers are provided and append to creation options
+    if input_data["creation_options"].get("layers"):
+        input_data["creation_options"]["layers"] = request.layers
     background_tasks.add_task(
         append_default_asset, dataset, version, input_data, default_asset.asset_id
     )
 
     # We now want to append the new uris to the existing ones and update the asset
     update_data = {"creation_options": deepcopy(default_asset.creation_options)}
     update_data["creation_options"]["source_uri"] += request.source_uri
+    if input_data["creation_options"].get("layers"):
+        update_data["creation_options"]["layers"] += request.layers
```

Comment:
Also, if request.layers is None, seems weird to append None to the layers list here, so maybe you want to also check in the `if` statement if request.layers is non-None?  Again, I might be mistunderstanding pydantic semantics here.

Reply:
Yeah, I think Dan is correct. Likewise, what happens if the user is appending a different source URI with a different layer name? Is it just listing all source URIs and layers? That's mostly fine but might be a little confusing to read.